config/peerpods/podvm/lib.sh

```diff
@@ -395,16 +400,134 @@ function get_image_type_url_and_path() {
     export PODVM_IMAGE_TYPE PODVM_IMAGE_URL PODVM_IMAGE_TAG PODVM_IMAGE_SRC_PATH
 }
 
+# Function to get format of the podvm image
+# Input: podvm image path
+# Use qemu-img info to get the image info
+# export the image format as PODVM_IMAGE_FORMAT
+function get_podvm_image_format() {
+    image_path="${1}"
+    echo "Getting format of the PodVM image: ${image_path}"
+
+    # jq -r when you want to output plain strings without quotes. Otherwise the string will be quoted
+    PODVM_IMAGE_FORMAT=$(qemu-img info -f raw --output json "${image_path}" | jq -r '.format') ||
+        error_exit "Failed to get podvm image info"
+
+    # vhd images are also raw format. So check the file extension. It's crude but for
+    # now it's good enough hopefully
+    if [[ "${image_path}" == *.vhd ]] && [[ "${PODVM_IMAGE_FORMAT}" == "raw" ]]; then
+        PODVM_IMAGE_FORMAT="vhd"
+    fi
+
+    echo "PodVM image format for ${image_path}: ${PODVM_IMAGE_FORMAT}"
+    export PODVM_IMAGE_FORMAT
+}
+
 # Function to validate the podvm image type.
+# Input: podvm image path
 function validate_podvm_image() {
-    PODVM_IMAGE_PATH="${1}"
+    image_path="${1}"
+
+    echo "Validating PodVM image: ${image_path}"
 
-    # Currently only qcow2 based PodVM images are supported for image upload.
-    if [[ "$(file -b "$PODVM_IMAGE_PATH")" != *QCOW2* ]]; then
-        error_exit "PodVM image is not a valid qcow2, exiting."
+    # Get the podvm image format. This sets the PODVM_IMAGE_FORMAT global variable
+    get_podvm_image_format "${image_path}"
+
+    # Check if the format is qcow2, raw or vhd
+    if [[ "${PODVM_IMAGE_FORMAT}" != "qcow2" &&
+        "${PODVM_IMAGE_FORMAT}" != "raw" &&
+        "${PODVM_IMAGE_FORMAT}" != "vhd" ]]; then
+        error_exit "PodVM image is neither a valid qcow2, raw or vhd, exiting."
     fi
 
-    echo "Checksum of the PodVM image: $(sha256sum "$PODVM_IMAGE_PATH")"
+    echo "Checksum of the PodVM image: $(sha256sum "$image_path")"
+}
+
+# Function to convert qcow2 image to vhd image
+# Input: qcow2 image
+# Output: vhddisk image
+function convert_qcow2_to_vhd() {
+    qcow2disk=${1}
+    rawdisk="$(basename -s qcow2 "${1}")raw"
+    vhddisk="$(basename -s qcow2 "${1}")vhd"
+    echo "Qcow2 disk name: ${qcow2disk}"
+    echo "Raw disk name: ${rawdisk}"
+    echo "VHD disk name: ${vhddisk}"
+
+    # Convert qcow2 to raw
+    qemu-img convert -f qcow2 -O raw "${qcow2disk}" "${rawdisk}" ||
+        error_exit "Failed to convert qcow2 to raw"
+
+    # Convert raw to vhd
+    resize_and_convert_raw_to_vhd_image "${rawdisk}"
```

Comment:
why going through RAW? for the alignment?  ` -o subformat=fixed,force_size` doesn't suppose to handle that?

Reply:
Right.. 1MB alignment needs to be handled separately.